homeassistant/components/bsblan/config_flow.py

```diff
@@ -0,0 +1,137 @@
+"""Config flow for BSB-Lan integration."""
+import logging
+from typing import Any, Dict, Optional
+
+from bsblan import BSBLan, BSBLanError, Info
+import voluptuous as vol
+
+from homeassistant.config_entries import CONN_CLASS_LOCAL_POLL, ConfigFlow
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+from homeassistant.helpers import ConfigType
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_DEVICE_IDENT, DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class BSBLanFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a BSBLan config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_LOCAL_POLL
+
+    async def async_step_user(
+        self, user_input: Optional[ConfigType] = None
+    ) -> Dict[str, Any]:
+        """Handle a flow initiated by the user."""
+        if user_input is None:
+            return self._show_setup_form()
+
+        try:
+            info = await self._get_bsblan_info(
+                user_input[CONF_HOST], user_input[CONF_PORT]
+            )
+        except BSBLanError:
+            return self._show_setup_form({"base": "connection_error"})
+
+        # Check if already configured
+        await self.async_set_unique_id(info.device_identification)
+        self._abort_if_unique_id_configured()
+
+        return self.async_create_entry(
+            title=info.device_identification,
+            data={
+                CONF_HOST: user_input[CONF_HOST],
+                CONF_PORT: user_input[CONF_PORT],
+                CONF_DEVICE_IDENT: info.device_identification,
+            },
+        )
+
+    async def async_step_zeroconf(
+        self, user_input: Optional[ConfigType] = None
+    ) -> Dict[str, Any]:
```

Comment:
```suggestion
    async def async_step_zeroconf(
        self, discovery_info: DiscoveryInfoType
    ) -> Dict[str, Any]:
```

Reply:
Hi Martin, 

DiscoveryInfoType? Where do I define this?

Reply:
I can't find it in the helpers. Or am I looking in the wrong direction

Reply:
It's here:
https://github.com/home-assistant/core/blob/71c8fcee20c55536b33c3ee774c76c1795f37cd2/homeassistant/helpers/typing.py#L11